File: android/src/main/kotlin/cl/ceisufro/fluttermbtilesextractor/ExtractResult.kt
package cl.ceisufro.fluttermbtilesextractor


class ExtractResult(val code: Int, val data: String) {
    var tiles: ArrayList<Tile> = arrayListOf()
    var mbtilesMetadata: MBTilesMetadata? = null

    constructor(code: Int, data: String, tiles: ArrayList<Tile>) : this(code, data) {
        this.tiles = tiles
    }

    constructor(code: Int, data: String, mbTilesMetadata: MBTilesMetadata, tiles: ArrayList<Tile>) : this(code, data) {
        this.tiles = tiles
        this.mbtilesMetadata = mbTilesMetadata
    }

    fun toMap(): HashMap<String, Any?> {
        val map = hashMapOf<String, Any?>()
        map.put("code", code)
        map.put("data", data)
        map.put("metadata", mbtilesMetadata?.toMap())
        map.put("tiles", Tile.toMapList(tiles))
        return map
    }
}

File: ios/Classes/ExtractResult.swift
//
//  ExtractResult.swift
//  flutter_mbtiles_extractor
//
//  Created by LJaraCastillo on 09-07-18.
//

import Foundation
class ExtractResult:NSObject{
    let code:Int
    let data:String
    let metadata:MBTilesMetadata?
    var tiles:[Tile]
    
    init(code:Int, data:String) {
        self.code = code
        self.data = data
        self.tiles = []
    }
    
    init(code:Int, data:String, tiles:[Tile]) {
        self.code = code
        self.data = data
        self.tiles = tiles
    }

    init(code:Int, data:String, metadata:MBTilesMetadata, tiles:[Tile]) {
        self.code = code
        self.data = data
        self.metadata = metadata
        self.tiles = tiles
    }
    
    func toMap() -> NSDictionary {
        return [
            "code":self.code,
            "data":self.data,
            "metadata":self.metadata?.toMap(),
            "tiles":Tile.toMapList(list: self.tiles),
        ]
    }
}
